src/Ethereum/Signer.cpp

```diff
@@ -84,88 +85,134 @@ Data addressStringToData(const std::string& asString) {
     return asData;
 }
 
-std::shared_ptr<TransactionNonTyped> Signer::buildNonTyped(const Proto::SigningInput& input) {
+std::shared_ptr<TransactionBase> Signer::build(const Proto::SigningInput& input) {
     Data toAddress = addressStringToData(input.to_address());
     uint256_t nonce = load(input.nonce());
     uint256_t gasPrice = load(input.gas_price());
     uint256_t gasLimit = load(input.gas_limit());
-    assert(gasPrice != 0);
+    uint256_t maxInclusionFeePerGas = load(input.max_inclusion_fee_per_gas());
+    uint256_t maxFeePerGas = load(input.max_fee_per_gas());
     switch (input.transaction().transaction_oneof_case()) {
         case Proto::Transaction::kTransfer:
             {
-                auto transaction = TransactionNonTyped::buildNativeTransfer(
-                    nonce, gasPrice, gasLimit,
+                if (gasPrice != 0) { // legacy
```

Comment:
instead of having multiple properties I would suggest to use enum on transaction itself
```
enum TransactionGasType {
    case gasPrice(gasPrice)
    Eip1559(maxFeePerGas, maxInclusionFeePerGas)
}
```

or something similar

Reply:
That's possible, and useful, but that would also break legacy code with legacy fee.

Reply:
It would also break Json format of SigningInput; I rather keep it backwards compatible